mmdet/models/anchor_heads/guided_anchor_head.py

```diff
@@ -0,0 +1,493 @@
+        assert self.loc_focal_loss, 'only focal loss is supported in loc'
+        self.cls_sigmoid_loss = cls_sigmoid_loss
+        self.cls_focal_loss = cls_focal_loss
+
+        self.approx_generators = []
+        self.base_approx_generators = []
+        for anchor_base in self.anchor_base_sizes:
+            self.approx_generators.append(
+                AnchorGenerator(anchor_base, self.octave_scales,
+                                self.octave_ratios))
+            self.base_approx_generators.append(
+                AnchorGenerator(anchor_base, [self.octave_scales[0]], [1.0]))
+        # one anchor per location
+        self.num_anchors = 1
+        if self.cls_sigmoid_loss:
+            self.cls_out_channels = self.num_classes - 1
+        else:
+            self.cls_out_channels = self.num_classes
+
+        self._init_layers()
+
+    def _init_layers(self):
+        self.relu = nn.ReLU(inplace=True)
+        self.conv_loc = nn.Conv2d(self.feat_channels, 1, 1)
+        self.conv_shape = nn.Conv2d(self.feat_channels, self.num_anchors * 2,
+                                    1)
+        deformable_groups = 4
+        offset_channels = 3 * 3 * 2
+        self.conv_offset = nn.Conv2d(
+            self.num_anchors * 2,
+            deformable_groups * offset_channels,
+            1,
+            bias=False)
+        self.conv_adaption = DeformConv(
+            self.feat_channels,
+            self.feat_channels,
+            kernel_size=3,
+            padding=1,
+            deformable_groups=deformable_groups)
+        self.conv_cls = MaskedConv2d(
+            self.feat_channels, self.num_anchors * self.cls_out_channels, 1)
+        self.conv_reg = MaskedConv2d(self.feat_channels, self.num_anchors * 4,
+                                     1)
+
+    def init_weights(self):
+        normal_init(self.conv_cls, std=0.01)
+        normal_init(self.conv_reg, std=0.01)
+        # normal_init(self.extra_rpn_conv, std=0.01)
+
+        normal_init(self.conv_offset, std=0.1)
+        normal_init(self.conv_adaption, std=0.01)
+
+        bias_cls = bias_init_with_prob(0.01)
+        normal_init(self.conv_loc, std=0.01, bias=bias_cls)
+        normal_init(self.conv_shape, std=0.01)
+
+    def forward_single(self, x):
+        loc_pred = self.conv_loc(x)
+        shape_pred = self.conv_shape(x)
+        offset = self.conv_offset(shape_pred.detach())
+        x = self.relu(self.conv_adaption(x, offset))
+        if not x.requires_grad:
+            mask = loc_pred.sigmoid()[0] >= self.loc_filter_thr
+            cls_score = self.conv_cls.forward_test(x, mask)
+            bbox_pred = self.conv_reg.forward_test(x, mask)
+        else:
+            cls_score = self.conv_cls(x)
+            bbox_pred = self.conv_reg(x)
+        return cls_score, bbox_pred, shape_pred, loc_pred
+
+    def forward(self, feats):
+        return multi_apply(self.forward_single, feats)
+
+    def get_anchors(self, featmap_sizes, shape_preds, img_metas):
```

Comment:
This method can be split into `get_anchors` and `get_sampled_anchors`, which is easier understanding.